datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -317,16 +317,27 @@ fn rename_expressions(
         .into_iter()
         .zip(new_schema.fields())
         .map(|(old_expr, new_field)| {
-            if &old_expr.get_type(input_schema)? == new_field.data_type() {
-                // Alias column if needed
-                old_expr.alias_if_changed(new_field.name().into())
-            } else {
-                // Use Cast to rename inner struct fields + alias column if needed
+            // Check if type (i.e. nested struct field names) match, use Cast to rename if needed
+            let new_expr = if &old_expr.get_type(input_schema)? != new_field.data_type() {
                 Expr::Cast(Cast::new(
                     Box::new(old_expr),
                     new_field.data_type().to_owned(),
                 ))
-                .alias_if_changed(new_field.name().into())
+            } else {
+                old_expr
+            };
+            // Alias column if needed
+            let new_name = new_field.name();
+            match &new_expr {
+                Expr::Column(c) => {
+                    // If expr is a column reference, alias_if_changed would cause an aliasing if the old expr has a qualifier
```

Comment:
Always calling `alias_if_changed` was causing things like `.. table.a AS a` in the final project, which is just unnecessary and slightly annoying in some testing cases (but shouldn't matter much in reality since the name is the same).